### PR TITLE
Exclude by default content-types !== 'text/html'

### PR DESCRIPTION
**Pull request by belgattitude**
see https://github.com/php-middleware/phpdebugbar/issues/18

### CODE REVIEW

File: src/PhpDebugBarMiddleware.php [hunk 1]
```diff
@@ -25,11 +25,25 @@ class PhpDebugBarMiddleware implements MiddlewareInterface
...
+     * @var bool
+     */
+    protected $excludeNonHtmlContent;
+
+    /**
+     * PhpDebugBarMiddleware constructor.
+     * @param DebugBarRenderer $debugbarRenderer
+     * @param bool $excludeNonHtmlContent Whether to disable debugbar on content-types != 'test/html'
+     */
+    public function __construct(DebugBarRenderer $debugbarRenderer, $excludeNonHtmlContent=true)
```
Comment (snapshotpl) [inline src/PhpDebugBarMiddleware.php]: Default value change current behaviour. Can we stay as is by default?

File: src/PhpDebugBarMiddleware.php [hunk 2]
```diff
@@ -49,7 +63,10 @@ public function process(ServerRequestInterface $request, DelegateInterface $dele
 
         if ($this->isHtmlResponse($response)) {
             return $this->attachDebugBarToResponse($response);
+        } elseif ($this->excludeNonHtmlContent) {
```
Comment (snapshotpl) [inline src/PhpDebugBarMiddleware.php]: Missing test case

File: src/PhpDebugBarMiddleware.php [hunk 3]
```diff
@@ -25,11 +25,25 @@ class PhpDebugBarMiddleware implements MiddlewareInterface
...
     protected $debugBarRenderer;
 
-    public function __construct(DebugBarRenderer $debugbarRenderer)
+    /**
+     * @var bool
+     */
+    protected $excludeNonHtmlContent;
+
+    /**
+     * PhpDebugBarMiddleware constructor.
```
Comment (snapshotpl) [inline src/PhpDebugBarMiddleware.php]: Useless information - to remove

File: src/PhpDebugBarMiddleware.php [hunk 4]
```diff
@@ -25,11 +25,25 @@ class PhpDebugBarMiddleware implements MiddlewareInterface
...
     use DoublePassCompatibilityTrait;
 
+    /**
+     * @var DebugBarRenderer
+     */
     protected $debugBarRenderer;
 
-    public function __construct(DebugBarRenderer $debugbarRenderer)
+    /**
+     * @var bool
```
Comment (snapshotpl) [inline src/PhpDebugBarMiddleware.php]: Maybe will be better to have some policy for that? Sometimes you want to debug json, but not generated pdf. With `bool` it's not possible. We can provide some read implmentation and simple interface to provide own.

**Comment by snapshotpl** [comment]
And as you see tests are broken ;(

File: src/PhpDebugBarMiddleware.php (hunk 3 above)
Comment (belgattitude) [inline src/PhpDebugBarMiddleware.php]: phpstorm ;)

File: src/PhpDebugBarMiddleware.php (hunk 4 above)
Comment (belgattitude) [inline src/PhpDebugBarMiddleware.php]: Agree... not sure if I can get some time to think about it. We could add a special key '__debugbar' if it does not already exist .   But I believe, going content-type-aware will need some more refactoring.

**Comment by belgattitude** [comment]
@snapshotpl I've fixed the tests by setting default `excludeNonHtmlContent` to false... But then Json is broken again then.

I'm not sure that I can fix the current behaviour, would need probably some breaking changes, but for sure refactoring of the middleware. 

I let the P/R living here, but good if you can find a better fix

**Pull request closed without merging**